common/achievement/bucket.ts

```diff
@@ -1,27 +1,23 @@
 import moment from 'moment';
 import { BucketFormula, DefaultBucket, GenericBucketConfig, TimeBucket } from './types';
-import { getBucketContext, getRelationTypeAndId } from './util';
 
 type BucketCreatorDefs = Record<string, BucketFormula>;
 
 // Buckets are needed to pre-sort and aggregate certain events by types / a certain time window (e.g. weekly) etc.
 export const bucketCreatorDefs: BucketCreatorDefs = {
     default: {
-        function: async (): Promise<GenericBucketConfig<DefaultBucket>> => {
-            return await { bucketKind: 'default', buckets: [] };
+        function: (): GenericBucketConfig<DefaultBucket> => {
+            return { bucketKind: 'default', buckets: [] };
         },
     },
     by_lecture_start: {
-        function: async (relation): Promise<GenericBucketConfig<TimeBucket>> => {
-            const [type] = getRelationTypeAndId(relation);
+        function: (bucketContext): GenericBucketConfig<TimeBucket> => {
+            const { context } = bucketContext;
```

Comment:
@dhenkel92 is that what you meant? that we pass a defined context to the bucket creator?

Reply:
Yes! Great 🙂